microbench/db_basic_bench.cc

```diff
@@ -129,6 +246,1019 @@ static void DBClose(benchmark::State& state) {
 BENCHMARK(DBClose)->Iterations(200);  // specify iteration number as the db size
                                       // is impacted by iteration number
 
+static void DBPut(benchmark::State& state) {
+  auto compaction_style = static_cast<CompactionStyle>(state.range(0));
+  uint64_t max_data = state.range(1);
+  uint64_t per_key_size = state.range(2);
+  bool enable_statistics = state.range(3);
+  bool enable_wal = state.range(4);
+  uint64_t key_num = max_data / per_key_size;
+
+  // setup DB
+  static DB* db;
+  Options options;
+  if (enable_statistics) {
+    options.statistics = CreateDBStatistics();
+  }
+  options.compaction_style = compaction_style;
+
+  auto rnd = Random(301 + state.thread_index);
+  KeyGenerator kg(&rnd, key_num);
+
+  if (state.thread_index == 0) {
+    SetupDB(state, options, &db, "DBPut");
+  }
+
+  auto wo = WriteOptions();
+  wo.disableWAL = !enable_wal;
+
+  for (auto _ : state) {
+    state.PauseTiming();
+    auto key = kg.Next();
+    auto val = rnd.RandomString(per_key_size);
+    state.ResumeTiming();
+    auto s = db->Put(wo, key, val);
+    if (!s.ok()) {
+      state.SkipWithError(s.ToString().c_str());
+    }
+  }
+
+  if (state.thread_index == 0) {
+    auto db_full = static_cast_with_check<DBImpl>(db);
+    auto s = db_full->WaitForCompact(true);
```

Comment:
This may be minor. According to https://google.github.io/styleguide/cppguide.html#Type_deduction which we follow, we usually use `auto` when it's clear what the type is. The `static_cast` cases fall into this category. For others, if the type is not that clear, we should still explicitly spell out the type.